### ISSUE TITLE
ScopeScreenScanner fails for stereo signals that cancels itself out.

### ISSUE THREAD

**Issue by elanhickler** [RobinSchmidt/RS-MET#64]
If you send it two opposing sinewaves stereo signal that cancels itself out (180 degrees out of phase) then you get silence and the scanner doesn't know what to do.

Should you think of a clever fix or should I add sync options: Sync L / Sync R / Sync Sum...

oh maybe you should have a switch within scopescanner for those options.

**Comment by RobinSchmidt** [comment]
yes, i have been leaving this to client code, what analysis signal to send into it. the class itself just accepts a mono signal for analysis. and then in both of our scopes, i feed it the sum of left and right. i think, for my class library, it's most flexible to leave this decision to client code. client code may want to use some kind of other "side-chain" signal, or a filtered input or whatever

**Comment by RobinSchmidt** [comment]
hmmm...not sure....maybe i could make a subclass that accepts stereo inputs and has a switch which channel to use (or even mix coefficients).

**Comment by RobinSchmidt** [comment]
just a random thought: might a nonlinear combination of the left and right channel signals be useful - such as the product? in case of a mono-signal, it would amount to squaring - with a little highpass, we'd still get the period (well, maybe multiplied by 2). i case of left = -right, it should work, too. ...well, really, as said - that's just a totally random idea...but might be worth to try

**Comment by elanhickler** [comment]
I don't know what would work, but in any case, it has to account for the signal going in and out of perfect phase or perfect cancellation. Otherwise just leave it alone and I'll put some options on PrettyScope for left/right/sum sync.

**Comment by elanhickler** [comment]
hmm, actually, MOTU oscilloscope is superior to your ScreenScanner in a few ways, more stable image and it can handle stereo signals in/out of phase, so there's room for improvement, but I don't want this to distract from the modulation system you're working on.

**Comment by elanhickler** [comment]
I don't know what MOTU oscilloscope is doing, but one thing it might be doing is seeing which signal is easier to get a lock on, or what signal has a lower frequency, or something.

It is also somehow ensuring that the position/phase of the signal remains in the same place visually.

Ok, weird, so it seems to be locking onto the higher frequency signal, but only switches when the frequency has significantly changes between the two, so it's not just immediately whatever is higher or lower.
